cmd click on recent notebook #3840

kpaharelau opened the pull request:
Use window.open for recent items from control page

kpaharelau committed:
cmd click on recent notebook #3840

Use window.open for recent items from control page

scottdraves commented on core/src/main/web/app/helpers/helper.js:
```diff
@@ -205,6 +205,28 @@
       getBaseUrl: function() {
         return bkUtils.getBaseUrl();
       },
+      openNotebookInNewWindow: function (notebookUri, uriType, readOnly, format) {
```
this can't be right we need a clean way.

kpaharelau committed:
cmd click on recent notebook #3840

>this can't be right we need a clean way.
DONE

kpaharelau committed:
cmd click on recent notebook #3840

I have used better URL quoting function

window.encodeURIComponent(notebookUri)

tomca32 commented on core/src/main/web/app/helpers/helper.js:
```diff
@@ -205,6 +205,22 @@
+
+        var path = "/open?uri=" + window.encodeURIComponent(notebookUri);
+        if (uriType) {
+          path += "&type=" + uriType;
```
No need to do this manually, you can use $httpParamSerializer.

https://docs.angularjs.org/api/ng/service/$httpParamSerializer

Docs are not great, here's an SO example:

http://stackoverflow.com/a/34039330/2341467

tomca32 commented on core/src/main/web/app/helpers/helper.js:
```diff
@@ -205,6 +205,22 @@
       },
+      openNotebookInNewWindow: function (notebookUri, uriType, readOnly, format) {
+
+        var path = "/open?uri=" + window.encodeURIComponent(notebookUri);
```
Since $httpParamSerializer will do the encoding, you can use it for uri param also.

kpaharelau committed:
cmd click on recent notebook #3840

>code is clean now, but the new window functionality should only apply if the cmd key is down (ctrl on pc), just like for normal links.
DONE

use $httpParamSerializer.
